readthedocs/core/permissions.py

```diff
@@ -132,14 +132,31 @@ def admins(cls, obj):
             return obj.owners.all()
 
     @classmethod
-    def members(cls, obj):
-        from readthedocs.organizations.models import Organization
+    def members(cls, obj, user=None):
+        """
+        Return the users that are members of `obj`.
+
+        If `user` is provided, we return the members of `obj` that are visible to `user`.
+        For a project this means the users that have access to the project,
+        and for an organization, this means the users that are on the same teams as `user`,
+        including the organization owners.
+        """
+        from readthedocs.organizations.models import Organization, Team
         from readthedocs.projects.models import Project
 
         if isinstance(obj, Project):
-            return obj.users.all()
+            project_owners = obj.users.all()
+            if user and user not in project_owners:
+                return User.objects.none()
```

Comment:
I am assuming this wouldn't affect the public project maintainer listing, correct? I don't see us passing in `user` there at least.

Reply:
Nope, when viewing a profile, we have a check to make sure we apply filtering on .com only, and we list maintainers from the template on the project detail page, that's overridden on .com to not show that.

Reply:
Did you also test this on the two beta dashboards? Some of the logic and queryset usage is a little different there. We don't need to hold this change up with any more fixes there, we can address these separately -- just wondering if we have more work here.

Reply:
Beta dashboard makes use of the same querysets, so all good there.